src/lib/parser.ts

```diff
@@ -0,0 +1,25 @@
+import * as ICAL from 'ical.js';
+import { FullCalendarEventObject } from '../types/caldav-parser';
+
+export function parse (ics: string): FullCalendarEventObject[] {
+  const jcal = ICAL.parse(ics);
+  const component: ICAL.Component = new ICAL.Component(jcal);
+  const subComponents: ICAL.Component[] = component.getAllSubcomponents();
+
+  return subComponents.map(componentToEvent);
+}
+
+
+const componentToEvent = (component: ICAL.Component): FullCalendarEventObject => {
+  const event: ICAL.Event = new ICAL.Event(component);
+  const fullcalendarEvent: FullCalendarEventObject = {
+    id: event.uid,
+    allDay: false,
```

Comment:
Shouldn't we have some comments about this?

Reply:
ye not finished yet :laughing: